### ISSUE TITLE
xone: find minimal initialization sequence

### ISSUE THREAD

**Issue by paroj** [paroj/xpad#20]
currently we use a 6 packet initialization sequencefor xbox one pads. ([obtained here](https://github.com/Fenisu/xbox/commit/6c4842e84ffbee3d2f5956ee5f7f4d7eae13b50c))

```c++
{{ 0x04, 0x20, 0x01, 0x00 }, 4, true},
{{ 0x01, 0x20, 0x01, 0x09, 0x00, 0x04, 0x20, 0x3a, 0x00, 0x00, 0x00, 0x80, 0x00}, 13, true},
{{ 0x01, 0x20, 0x01, 0x09, 0x00, 0x04, 0x20, 0xba, 0x00, 0x00, 0x00, 0x00, 0x00}, 13, true},
{{ 0x05, 0x20, 0x02, 0x09, 0x06, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x55, 0x53}, 13, true},
{{ 0x05, 0x20, 0x03, 0x01, 0x00}, 5, true},
{{ 0x0a, 0x20, 0x04, 0x03, 0x00, 0x01, 0x14}, 7, true}
```

However it originates from a dump of a single controller on Windows. We need to figure out their meaning and test whether all of them are really needed.

**Comment by paroj** [comment]
somebody with a xbox one pad (FW 2015) should uncomment the lines one by one until the controller stops working and report back the needed combination. Do not forget to plug the controller in and out between each try.

**Comment by socketbind** [comment]
It seems that only the following initialization packets are needed:
```
struct xpad_output_packet xpad_one_init[] = {
	/* start controller w/o input */
	{{ 0x04, 0x20, 0x01, 0x00 }, 4, true},
	/* this was the only packet needed for pre 2015 firmware*/
	{{ 0x05, 0x20, 0x03, 0x01 /* rumble enable? */, 0x00}, 5, true} // required
};
```

The controller lights up and all buttons/axes work. Removing the first packet will make the controller unresponsive and not send any input. Could somebody please verify this for me with their own hardware?

I used the following testing sequence:
1. Unplug controller
2. rmmod xpad
3. Comment out an element in xpad_one_init
4. sudo dkms remove -m xpad -v 0.4 --all
5. sudo dkms install -m xpad -v 0.4
6. modprobe xpad
7. Plug in the controller and verify with jstest

**Comment by LiranV** [comment]
Just tested it too with the same two packets @socketbind used and it works.

**Comment by paroj** [comment]
could you do one more test with
```c++
static const struct xpad_output_packet xpad_one_init[] = {
	{{ 0x05, 0x20, 0x03, 0x01 /* rumble enable? */, 0x00}, 5, true},
	{{ 0x05, 0x20, 0x03, 0x01 /* rumble enable? */, 0x00}, 5, true},
};
```
i.e. verify that we really need that specific packet and not an arbitrary initialization packet

**Comment by socketbind** [comment]
I replaced the array with the one you provided and it also seems to fix the issue.

**Comment by paroj** [comment]
sorry for the noise - I chose the packets poorly as `0x4 & 0x5 == 1`. Could you try:

```c++
static const struct xpad_output_packet xpad_one_init[] = {
	{{ 0x01, 0x20, 0x01, 0x09, 0x00, 0x04, 0x20, 0x3a, 0x00, 0x00, 0x00, 0x80, 0x00}, 13, true},
	/* this was the only packet needed for pre 2015 firmware*/
	{{ 0x05, 0x20, 0x03, 0x01 /* rumble enable? */, 0x00}, 5, true},
};
```

**Comment by LiranV** [comment]
@paroj This works for me.
Tested with jstest.

**Comment by socketbind** [comment]
Works great with my controller too.

**Comment by paroj** [comment]
hmm - so far this does not clarify much.

could you test again using the master branch (as of now). Initially there was a bug such that the first packet was not submitted - maybe that was all..

**Comment by socketbind** [comment]
Recompiled it, making it sure that it contains the "back to start.." commit and the controller is working properly with the master branch.

**Comment by socketbind** [comment]
I'm still curious though which commit fixed the actual issue because when I reverted the array to its original state (with only the original init sequence) the controller stopped working. I think I was using the xone2015 branch but I accidentally managed to wipe it.

**Comment by paroj** [comment]
I force push changes to keep the number of commits low, therefore the broken version wiped from history.

However you this is the fix we actually needed here: c27e6369098b3f5ebbd11f14c0524aa5f9a8c20a

**Comment by dantob** [comment]
the version is master is working, but controller is once again detected as special forces. You left out this part 

-	{ 0x045e, 0x02dd, "Microsoft X-Box One pad (Covert Forces)", 0, XTYPE_XBOXONE },
+	{ 0x045e, 0x02dd, "Microsoft X-Box One pad (Firmware 2015)", 0, XTYPE_XBOXONE },
